lib/repl.js

```diff
@@ -182,14 +182,13 @@ function REPLServer(prompt, stream, eval_, useGlobal, ignoreUndefined) {
   self.bufferedCommand = '';
   self.lines.level = [];
 
-  function complete(text, callback) {
-    self.complete(text, callback);
-  }
+  // Figure out which "complete" function to use.
+  self.completer = options.completer || complete;
```

Comment:
Can you change this to `self.completer = typeof options.completer === 'function' ? options.completer : complete;` just to ensure that `completer` is a function.